src/flatten_dict/flatten_dict.py

```diff
@@ -81,6 +83,7 @@ def flatten(
 
     if isinstance(reducer, str):
         reducer = REDUCER_DICT[reducer]
+    reducer_accepts_parent_obj = len(inspect.signature(reducer).parameters) == 3
```

Comment:
python 2.7 doesn't have `inspect.signature`, so you need to use (funcsigs)[https://funcsigs.readthedocs.io/en/0.4/]. You can refer to how I did for `pathlib`. I know it's annoying. If you don't want to dig into the poetry config. You can try-catch this line and give `reducer_accepts_parent_obj` a default value `False`, and I will refactor this to support python 2.7 later.

Reply:
Oh, but the test will fail if you do so. You can use (pytest.mark.skipif)[https://docs.pytest.org/en/latest/how-to/skipping.html#id1]